code/game/turfs/turf.dm

```diff
@@ -728,3 +737,41 @@ GLOBAL_LIST_EMPTY(station_turfs)
 /// Allows for reactions to an area change without inherently requiring change_area() be called (I hate maploading)
 /turf/proc/on_change_area(area/old_area, area/new_area)
 	transfer_area_lighting(old_area, new_area)
+
+//A check to see if graffiti should happen
+/turf/proc/try_graffiti(mob/vandal, obj/item/tool)
+
+	if(!tool.sharpness)
+		return FALSE
+
+	if(!vandal.canUseTopic(src, USE_CLOSE) || !vandal.is_holding(tool))
+		return FALSE
+
+	if(HAS_TRAIT_FROM(src, TRAIT_NOT_ENGRAVABLE, INNATE_TRAIT))
+		to_chat(vandal, span_warning("[src] cannot be engraved!"))
+		return FALSE
+
+	if(HAS_TRAIT_FROM(src, TRAIT_NOT_ENGRAVABLE, TRAIT_GENERIC))
+		to_chat(vandal, span_warning("[src] already has an engraving."))
+		return FALSE
+
+	var/message = stripped_input(vandal, "Enter a message to engrave.", "Engraving", null ,64, TRUE)
+	if(!message)
+		return FALSE
+
+	if(!vandal.canUseTopic(src, USE_CLOSE) || !vandal.is_holding(tool))
+		return FALSE
```

Comment:
```suggestion
		return TRUE
```
Since we've started an action, even if it's interrupted, it should cancel propagation.